socketapi.js

```diff
@@ -19,31 +34,63 @@ io.on("connection", (socket) => {
         }
         nextIds[code]++;
         // send board and next id back to user
-        socket.emit("joindata", {nextId: nextIds[code], board: boards[code]})
+        socket.emit("joinData", {nextId: nextIds[code], board: boards[code]})
+    });
+
+    //  drawPen
+    //  Broadcast client drawing with the pen
+    //      data: 
+    //      {
+    //          code: the code for the board being drawn on
+    //          id: the id of the pen object
+    //          size: the stroke-width of the pen object
+    //          color: the color of the pen object
+    //          newPoints: the points being added to the Pen object's path
+    //      }
+    socket.on("drawPen", function(data) {
+        socket.to(data.code).emit('drawPen', data)
     });
 
-    socket.on("add", function(data) {
-        boards[data.code][data.id] = data;
-        socket.to(data.code).emit('add', data)
+    //  addPen
+    //  Add a finished Pen object to the saved board for sending to client on join
+    //      data: 
+    //      {
+    //          code: the code for the board being drawn on
+    //          id: the id of the pen object
+    //          size: the stroke-width of the pen object
+    //          color: the color of the pen object
+    //          path: the path of the pen object (to make the SVG)
+    //      }
+    socket.on("addPen", function(data) {
+        // set the board object
+        boards[data.code][data.id] = {type:"Pen", data: {path: data.path, size: data.size, color:data.color}};
     });
 
     socket.on("updateTransform", function(data) {
 
     });
-
+    
+    //  requestNewId
+    //  Get a new ID for the client to attach to a new drawing object, done
+    //    on server side so that each user's drawings increments the nextId.
+    //      data: 
+    //      {
+    //          code: the code for the board being drawn on
+    //      }
     socket.on("requestNewId", function(data) {
```

Comment:
I noticed that the order in which drawings appear is based on the highest ID value. For example, if user A and user B connect, A will have drawing ID 1 and B will have drawing ID 2. When A draws, their drawing ID is updated to 3. Let's say A draws again and now the board has two objects with IDs 1 and 3. When B draws, they create an object with ID 2 which will be displayed below object with ID 3. We might need to rethink how to update IDs (Maybe send a new ID with a higher value to everyone once something is added).